src/custom-element.js

```diff
@@ -976,6 +996,9 @@ function createBaseCustomElementClass(win) {
      * @private @final @this {!Element}
      */
     tryUpgrade_() {
+      if (this.isAwaitingSize_()) {
```

Comment:
While an element is in the awaiting-size state it has different properties than extensions may be assuming (i.e. children display: none, overflow: visible). So it seems safest to hold-off upgrading until the element is back in the normal state (i.e. it's size-defined).